### PR TITLE
[9. GET api/reviews/review_id/comments] adds endpoint to retrieve all…

### PR DESCRIPTION
**Pull request by Proto-Will**
… comments related to review_id

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -223,33 +223,58 @@ describe('app', () => {
...
+                review_id: expect.any(Number),
+                author: expect.any(String),
+                votes: expect.any(Number),
+                created_at: expect.any(String)
+              })
+            )
           })
         })
       })
+  })
```
Comment (emilyb93) [inline __tests__/api.test.js]: No error testing here, i want to see test for anything that the user can do wrong.
- What if the user tries to get a review that doesnt exist?
- what if the review does exist but theres no comments?
- what if the review_id requesting is not a number?

**Review by emilyb93 — CHANGES_REQUESTED** [review]
More testing please

File: __tests__/api.test.js [hunk 2]
```diff
@@ -251,5 +251,56 @@ describe('app', () => {
...
+        .then(({ body }) => {
+          const { comments } = body;
+          expect(comments).toBeInstanceOf(Array);
+          expect(comments).toHaveLength(3);
+          comments.forEach((comment) => {
+            expect(comment).toEqual(
+              expect.objectContaining({
+                comment_id: expect.any(Number),
+                body: expect.any(String),
+                review_id: expect.any(Number),
```
Comment (CatHoang) [inline __tests__/api.test.js]: You could hardcode this review_id in as we know it would be 3

File: __tests__/api.test.js [hunk 3]
```diff
@@ -251,5 +251,56 @@ describe('app', () => {
...
+        .then(({body: { msg }}) => {
+          expect(msg).toBe('No comment found for review_id: 50')
+        })
+      })
+      test('404; handles id with no comments', () => {
+        return request(app)
+        .get('/api/reviews/1/comments')
+        .expect(404)
+        .then(({body: { msg }}) => {
+          expect(msg).toBe('No comment found for review_id: 1')
```
Comment (CatHoang) [inline __tests__/api.test.js]: would we expect an 200 for an article that exists but has no comments? Think about how this information would come back, maybe as an empty array?

**Review by CatHoang — COMMENTED** [review]
Hi Will, just a couple of suggestions :)

File: __tests__/api.test.js [hunk 4]
```diff
@@ -251,5 +251,56 @@ describe('app', () => {
...
+        })
+      })
+      test('204; handles incorrect id', () => {
+        return request(app)
+        .get('/api/reviews/50/comments')
+        .expect(204)
+        .then(({body: { msg }}) => {
+          expect(msg).toBe(undefined)
+        })
+      })
```
Comment (Dutton1990) [inline __tests__/api.test.js:286]: This should be a 404 as the id is valid but the review doesn't exist, you can deal with this by checking the rows/rowCount after the query and returning a Promise.reject with your error message.

File: __tests__/api.test.js [hunk 5]
```diff
@@ -251,5 +251,56 @@ describe('app', () => {
...
+      test('204; handles id with no comments', () => {
+        return request(app)
+        .get('/api/reviews/1/comments')
+        .expect(204)
+        .then(({body: { msg }}) => {
+          expect(msg).toBe(undefined)
+        })
```
Comment (Dutton1990) [inline __tests__/api.test.js:293]: This should be a 200 as you would still want to return an empty array of comments for a review that does exist. 

You will have to make another query in your model after checking the rowCount/rows. Check whether you can select this article from the database. If you can, return an empty array. If it doesn't exist, return an error.

**Pull request merged**